src/main/java/net/imagej/ops/features/sets/AbstractConfigurableComputerSet.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * @author Tim-Oliver Buchholz, University of Konstanz
+ *
+ * @param <I>
+ *            type of the common input
+ * @param <O>
+ *            type of the common output
+ */
+public abstract class AbstractConfigurableComputerSet<I, O extends Type<O>> extends AbstractComputerSet<I, O>
+		implements ConfigurableComputerSet<I, O> {
+
+	@Parameter(required = false)
+	private List<Class<? extends Op>> active;
+
+	/**
+	 * The activated {@link Computers}.
+	 */
+	private final Map<Class<? extends Op>, Boolean> activated;
+
+	/**
+	 * Create a new {@link AbstractConfigurableComputerSet} with the default
+	 * {@link OpEnvironment}.
+	 *
+	 * @param outputTypeInstance
+	 *            object of the output type
+	 * @param inType
+	 *            the input type
+	 */
+	public AbstractConfigurableComputerSet(final O outputTypeInstance, final Class<I> inType) {
+		super(null, outputTypeInstance, inType);
+		active = new ArrayList<>();
+		activated = new HashMap<>();
+	}
+
+	/**
+	 * Create a new {@link AbstractConfigurableComputerSet} with a custom
+	 * {@link OpEnvironment}.
+	 *
+	 * @param opEnv
+	 *            the custom {@link OpEnvironment}
+	 * @param outputTypeInstance
+	 *            object of the output type
+	 * @param inType
+	 *            the input type
+	 */
+	public AbstractConfigurableComputerSet(final OpEnvironment opEnv, final O outputTypeInstance,
+			final Class<I> inType) {
+		super(opEnv, outputTypeInstance, inType);
+		active = new ArrayList<>();
+		activated = new HashMap<>();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void initialize() {
+		if (active.isEmpty()) {
+			active.addAll(Arrays.asList(getFeatures()));
+		}
+		super.initialize();
+	}
+
+	/**
+	 * Set {@link CustomOpEnvironment} and create all computers of this
+	 * featureset and activate passed {@link Computers}.
+	 *
+	 * @param infos
+	 *            for the {@link CustomOpEnvironment}
+	 */
+	@Override
+	protected void initialize(final Collection<OpInfo> infos) {
+		if (infos != null) {
+			setEnvironment(new CustomOpEnvironment(ops(), infos));
+		}
+
+		for (final Class<? extends Op> feature : getFeatures()) {
+			if (active.contains(feature)) {
+				activated.put(feature, true);
+				addComputer(feature);
+			} else {
+				activated.put(feature, false);
+			}
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Map<String, O> compute1(final I input) {
+		for (final UnaryComputerOp<I, O> computer : activated.entrySet().stream().filter(a -> a.getValue())
+				.map(a -> computers.get(a.getKey())).collect(Collectors.toList())) {
+			computer.compute1(input, computer.out());
+		}
+
+		return namedOutputs;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean isActive(final Class<? extends Op> feature) {
+		return activated.get(feature);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Class<? extends Op>> getActive() {
+		return activated.entrySet().stream().filter(e -> e.getValue()).map(e -> e.getKey())
```

Comment:
This should return all Computers which were computed. In case of a ComputerSet it returns just all Computers in case of a ConfigurableComputerSet it returns just the active ones. I could move getActive() from ConfigurableComputerSet to ComputerSet and just call getActive().